src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -1196,7 +1199,7 @@ private void HandleReadResponseContentException(Exception ex, CancellationToken
                     // Our connection was reset. Start aborting the connection.
                     Exception abortException = _connection.Abort(ex);
                     throw new IOException(SR.net_http_client_execution_error, new HttpRequestException(SR.net_http_client_execution_error, abortException));
```

Comment:
I'm wondering why are we throwing `IOException` below? Aren't these errors supposed to surface as `HttpProtocolException` from `HttpContent`'s `Http3ReadStream` (which seems to delegate to this method)?